derivative.go

```diff
@@ -140,7 +141,7 @@ func (n *DerivativeNode) derivative(prev, curr models.Fields, prevTime, currTime
 	elapsed := float64(currTime.Sub(prevTime))
 	if elapsed == 0 {
 		n.incrementErrorCount()
-		n.logger.Printf("E! cannot perform derivative elapsed time was 0")
+		n.diag.Error("cannot perform derivative", errors.New("elaspsed time was 0"))
```

Comment:
Can the n.diag.Error() method call n.incrementErrorCount() internally? That would remove the onus on each node to correctly count their errors and rather only need to report them to the Diagnostic interface. This probably means you need a small shim implementation of the interface within the kapacitor package to intercept Error and With* calls.

Does that sound reasonable and useful?

Reply:
Yeah, I thought about doing something like that, but I wasn't sure exactly the best way to do that.

Reply:
fixed